projs/shadow/shadow-engine/core/inc/shadow/core/CoreModule.h

```diff
@@ -1,25 +1,25 @@
-#pragma once
-
-#include "Module.h"
-
-class CoreModule : public ShadowEngine::Module {
-  SHObject_Base(CoreModule)
-  public:
-    CoreModule() : Module() {}
-
-    void PreInit() override {
-
-    }
-
-    void Init() override {
-
-    }
-
-    void Update(int frame) override {
-
-    }
-
-    void Destroy() override {
-
-    }
-};
+#pragma once
+
+#include "Module.h"
+
+class CoreModule : public SH::Module {

```

Comment:
For now, nothing It is just a marker module to see that the engine's core has been loaded.
We can put some Library, System, etc info in here. 
It could also be merged with the Application but that introduces a circle.